limit-types/README.md

```diff
@@ -0,0 +1,333 @@
+# Limit types
+
+> You need to install the [`buf` CLI][install] to follow along with this example. Requires v1.1.0 or
+> above.
+
+The [`buf build`][build] command creates a [Buf image][image] (or
+[`FileDescriptorSet`][filedescriptorset]) out of a [Buf input][input]. By default, the image or
+`FileDescriptorSet` includes _all_ of the Protobuf types—messages, enums, and services—in the input.
+But the `buf build` command offers [`--type`][type] option that enables you to supply fully
+qualified Protobuf type names and limit the output to only the Protobuf descriptors required to
+represent those types and their required dependencies.
+
+In this example, we can use `buf build --type` to view different representations (as JSON) for
+various Protobuf types in the [`auth/v1`](./auth/v1) module.
+
+The `auth/v1` module has a [`User`](./auth/v1#L5-9) message with no dependencies. To display a JSON
+representation of this type:
+
+```sh
+buf build \
+  --type auth.v1.User \
+  --output -#format=json | \
+  jq '.file[0].messageType[0]'
```

Comment:
With `[0]` you're explictly saying you only want to look at the first message type of the first file, which doesn't give complete clarity on the flag (since you filter the output you're showing, how do I know the original buf output didn't contain more?)

A jq query like `'.file[] | {name: .name, messages: .messageType | map(.name)}'` will for each file in the output show you its names and the names of the messages, which better shows that there are no other dependencies.